simulation/amaru-sim/src/simulator/run.rs

```diff
@@ -315,3 +284,72 @@ impl BlockFetcher for FakeBlockFetcher {
         Ok(vec![])
     }
 }
+
+/// This implementation of ForwardEventListener sends the received events a Sender that can collect them
+/// to check them later.
+///
+/// A message id is maintained for each received event and each message is duplicated to the number of downstream peers.
+#[derive(Clone, Debug)]
+pub struct MockForwardEventListener {
+    node_id: String,
+    number_of_downstream_peers: u8,
+    sender: mpsc::Sender<Envelope<ChainSyncMessage>>,
+    msg_id: Arc<Mutex<u64>>,
+}
+
+impl MockForwardEventListener {
+    pub fn new(
+        node_id: String,
+        number_of_downstream_peers: u8,
+        sender: mpsc::Sender<Envelope<ChainSyncMessage>>,
+    ) -> Self {
+        Self {
+            node_id,
+            number_of_downstream_peers,
+            msg_id: Arc::new(Mutex::new(0)),
+            sender,
+        }
+    }
+}
+
+#[async_trait]
```

Comment:
this is no longer necessary since a couple of Rust versions

Reply:
If I don't use the annotation I'm getting an error telling me that "the trait `ForwardEventListener` is not dyn compatible". I could revert to using a boxed future but since that what the annotation does under the hood, we might as well keep it, right?